test/suites/parathreads/test_tanssi_parathreads.ts

```diff
@@ -76,65 +74,193 @@ describeSuite({
+                    }
+                }
+
+                const currentBlockNumber = (await relayApi.rpc.chain.getHeader()).toJSON().number;
+                const txsToRemoveCoretime = coresToRemoveFromCoretime.map((core) => {
+                    return relayApi.tx.coretime.assignCore(
+                        core,
+                        currentBlockNumber + 10,
+                        [[{ Pool: null }, 57600]],
+                        null
+                    );
+                });
+
+                await signAndSendAndInclude(
+                    relayApi.tx.sudo.sudo(relayApi.tx.utility.batch(txsToRemoveCoretime)),
+                    alice
+                );
+            },
+        });
+
+        it({
+            id: "T01b",
+            title: "Fund parachain tank account on relay chain",
+            test: async function () {
+                const keyring = new Keyring({ type: "sr25519" });
+                const alice = keyring.addFromUri("//Alice", { name: "Alice default" });
+
+                // Hard coded tank accounts; we still need to find a way to generate them
+                const paraId2000Tank = "5Eyq93LQvDWbHmncBeXcMcDvyRKShc2ywpFUzhCX7AcJfsxm";
+                const paraId2001Tank = "5GhKA47pfc1XWbMGbAYoP3F8Lpnvodc2M71NC7bgJH8JGbqB";
+
+                const aliceBalance = (await relayApi.query.system.account(alice.address)).data.free.toBigInt();
+                const balanceToTransfer = aliceBalance / 4n;
+
+                const batchTx = relayApi.tx.utility.batch([
+                    relayApi.tx.balances.transferKeepAlive(paraId2000Tank, balanceToTransfer),
+                    relayApi.tx.balances.transferKeepAlive(paraId2001Tank, balanceToTransfer),
+                ]);
+                await signAndSendAndInclude(batchTx, alice);
+            },
+        });
+
+        it({
+            id: "T02a",
+            title: "Disable full_rotation and set xcm weights",
+            timeout: 6000000,
             test: async function () {
                 const keyring = new Keyring({ type: "sr25519" });
                 const alice = keyring.addFromUri("//Alice", { name: "Alice default" });
                 const tx4 = await paraApi.tx.configuration.setFullRotationPeriod(0);
                 await signAndSendAndInclude(paraApi.tx.sudo.sudo(tx4), alice);
+
+                const tx = paraApi.tx.sudo.sudo(
+                    paraApi.tx.xcmCoreBuyer.setRelayXcmWeightConfig({
+                        buyExecutionCost: 50_000_000_000,
+                        weightAtMost: {
+                            refTime: 1_000_000_000,
+                            proofSize: 100_000,
+                        },
+                    })
+                );
+                const setRelayChainTx = paraApi.tx.sudo.sudo(paraApi.tx.xcmCoreBuyer.setRelayChain("Rococo"));
+                await signAndSendAndInclude(
+                    paraApi.tx.sudo.sudo(paraApi.tx.utility.batch([tx, setRelayChainTx])),
```

Comment:
`signAndSendAndInclude` is a bit slow to call because it waits for the tx to be finalized, which can take 30-60 seconds, so if possible try to batch as many extrinsics as possible into one call. So instead of 2 separate calls to `signAndSendAndInclude`, move `tx4` here.

```suggestion
                    paraApi.tx.sudo.sudo(paraApi.tx.utility.batch([tx4, tx, setRelayChainTx])),
```

It doesn't matter much so if it breaks just keep it